### ISSUE TITLE
API key inconsistency

### ISSUE THREAD

**Issue by bkazi** [toddmotto/public-apis#140]
I noticed that Musixmatch API notes that an apikey query string required in the OAuth coulmn
As per my experiences APIs such as MovieDB, Wunderground, Open Weather Map, Yahoo Weather and maybe others require similar apikey query strings with the request.
I was wondering if this is a choice that has been made or an inconsistency in the list

Thanks for the wonderful compilation!

**Comment by mikestreety** [comment]
Hi @bkazi - If an API key is required it should say so. If there are any missing this is just a mistake. 

Now that we've merged a load, are there any you know require a key which isn't specified?

**Comment by YannBertrand** [comment]
May we add a column to check if an api key is required (when not using OAuth)?

**Comment by mikestreety** [comment]
I think that issue has some validity - changing `oAuth` to `Auth` means the "No but an `apiKey` is required can be changed to just `apiKey`

**Comment by YannBertrand** [comment]
Current solution:

| API | Description | OAuth | Link |
|---|---|---|---|
| API 1 | API 1 | Yes | Go! |
| API 2 | API 2 | No, but an `apiKey` is required | Go! |
| API 3 | API 3 | No | Go! |

Rename-oauth-column solution:

| API | Description | *Auth* | Link |
|---|---|---|---|
| API 1 | API 1 | *`OAuth`* | Go! |
| API 2 | API 2 | *`apiKey`* | Go! |
| API 3 | API 3 | No | Go! |

Add-a-column solution:

| API | Description | *Auth* | OAuth | Link |
|---|---|---|---|---|
| API 1 | API 1 | *Yes* | Yes | Go! |
| API 2 | API 2 | *Yes* | No | Go! |
| API 3 | API 3 | *No* | No | Go! |

I agree with @mikestreety, the second solution seems to be the neatest. What do think of it?

**Comment by jbrooksuk** [comment]
Hmm, if we're to generate JSON from this data, then we want it to be as verbose as possible, right?

**Comment by mikestreety** [comment]
I don't think we should let potential JSON-ification get in the way of readability? We could head down route two and then route three for the JSON and compile it into a route 2 looking table?

**Comment by davemachado** [comment]
Because this list of API's is publicly supported, I think the renaming the OAuth column to just Auth adds the opportunity to add a bit more detail into the method of authorization instead of just a blunt Yes/No. I would recommend this in hopes of keeping a cosistant naming scheme so that we don't end up having variations of one type of Auth (apikey, ApiKey, Key, APIKEY, etc).

**Comment by mikestreety** [comment]
We could add a "contributing" guide and include a list of existing ones?

**Comment by davemachado** [comment]
To solve the issue of having multiple naming schemes for the same auth type, perhaps adding a table at the top of the README to explain the different choices for Auth would be best suited.

**Comment by davemachado** [comment]
@mikestreety beat me to it.... 👍

**Comment by mikestreety** [comment]
There is one PR outstanding. Once merged I say let's go for this 👍

**Comment by davemachado** [comment]
I just opened a PR that adds a contributing guide to the project. I think it pretty much outlines what we are looking for in a submission (including the apiKey inconsistency), but if anyone has changes they'd like to propose, I'd be more than happy to add them in. :)

**Comment by mikestreety** [comment]
This has also been resolved in the same PR :smile: Once merged we can finally get rid of this issue!

**Comment by mikestreety** [comment]
Closing this as it is now resolved
